### DISCUSSION TITLE
Non linear combinations in SAF

### DISCUSSION THREAD

**Discussion by sonkaste17** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#51] [Ideas]
Hello SAF community!

So far in SAF, we did not provide the possibility for defining the non linear combination. The proposal is to create a new value for attribute "Type" = "Non linear" with relevant "Category" = "ULS (Ultimate Limit State)", "SLS (Serviceability Limit State)" and "Not defined" in the object StructuralLoadCombination. Then provide a list of load cases in the specific combination. This structure can cooperate together with result 1D (ResultInternalForce1D) where results of the non linear analysis can be written down for non linear combinations.

**Review - Proposal for object StructuralLoadCombination**
| Attribute &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;   |      Values       |        Comment       |  
|:-----------------|:-------------:|:-------------:|
| Type |  Linear, Envelope, **Non linear** | Non linear is a new attribute for the type of combination | 
| Category |    ULS (Ultimate Limit State), SLS (Serviceability Limit State), Not defined   |  Categories relevant for Non linear type of combination | 

**Example:**
StructuralLoadCombination
![image](https://user-images.githubusercontent.com/81751485/142599350-dd0d44c3-caa6-4265-9f06-1245ad6caff3.png)

ResultInternalForce1D
![image](https://user-images.githubusercontent.com/81751485/142599733-77157230-e252-41e2-9d18-08bf86b4c0e3.png)

This approach provides the ability to define non linear combinations in SAF and possibly write results in the object ResultInternalForce1D. What is not covered in the definition are imperfections for the specific combination (the setup for non linear calculation). This can be written down to SAF to the new columns for every non-linear combination. Do you want to have in SAF also the definition of imperfection as input for your analysis software?

Do you think we are missing something?

Thank you,
Štěpán

**Comment by dschuermans** [comment]
At first glance I can't see any issues implementing this in the SDK, +1 from me

**Comment by MartinSalak** [comment]
The nonlinear combination is already published in the latest documentation, which will be released as the next SAF version. Please take a look, all comments and notes are welcome.
https://www.saf.guide/en/latest/loads/structuralloadcombination.html
